src/Tickets/Commerce/Order_Modifiers/API/Coupons.php

```diff
@@ -0,0 +1,540 @@
+			'payment_intent_id' => [
+				'description' => esc_html__( 'The payment intent to apply the coupon to.', 'event-tickets' ),
+				'type'        => 'string',
+				'format'      => 'text-field',
+				'required'    => true,
+			],
+			'purchaser_data'    => [
+				'description' => esc_html__( 'The purchaser data.', 'event-tickets' ),
+				'type'        => 'object',
+				'required'    => true,
+				'properties'  => [
+					'name'  => [
+						'description' => esc_html__( 'The purchaser name.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'text-field',
+						'required'    => true,
+					],
+					'email' => [
+						'description' => esc_html__( 'The purchaser email.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'email',
+						'required'    => true,
+					],
+				],
+			],
+		];
+
+		switch ( $endpoint ) {
+			case 'apply':
+			case 'remove':
+				return array_merge( $coupon_args, $common_args );
+
+			case 'validate':
+				return $coupon_args;
+
+			case 'create':
+				return [
+					'slug'         => [
+						'description' => esc_html__( 'The coupon slug.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'text-field',
+						'required'    => true,
+					],
+					'display_name' => [
+						'description' => esc_html__( 'The coupon display name.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'text-field',
+						'required'    => true,
+					],
+					'status'       => [
+						'description' => esc_html__( 'The coupon status.', 'event-tickets' ),
+						'type'        => 'string',
+						'enum'        => [ 'active', 'inactive', 'draft' ],
+						'required'    => true,
+					],
+					'amount'       => [
+						'description' => esc_html__( 'The coupon amount.', 'event-tickets' ),
+						'type'        => 'integer',
+						'required'    => true,
+					],
+					'sub_type'     => [
+						'description' => esc_html__( 'The coupon sub type.', 'event-tickets' ),
+						'type'        => 'string',
+						'enum'        => [ 'percent', 'flat' ],
+						'required'    => false,
+						'default'     => 'flat',
+					],
+					'start_time'   => [
+						'description' => esc_html__( 'The coupon start time.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'date-time',
+						'default'     => null,
+					],
+					'end_time'     => [
+						'description' => esc_html__( 'The coupon end time.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'date-time',
+						'default'     => null,
+					],
+				];
+
+			default:
+				return [];
+		}
+	}
+
+	/**
+	 * Get the purchaser information.
+	 *
+	 * @since TBD
+	 *
+	 * @param Request $request The request object.
+	 *
+	 * @return array
+	 */
+	protected function get_purchaser_information( Request $request ) {
+		$purchaser_data = $request->get_param( 'purchaser_data' );
+
+		[ $first_name, $last_name ] = explode( ' ', $purchaser_data['name'], 2 );
```

Comment:
It's required for all tickets purchases through TicketsCommerce

Reply:
> Is there any instance where it might not be set?

The `name` field inside `purchaser_data` is a required part of the schema. So it will always be present from the API parsing.